File: interviews/lc/12-reverse-string.py

def reverseStr(s: str) -> str:
    return "".join([c for c in reversed(s)])


def reverseList(s: str) -> str:
    lst = [ord(i) for i in s]
    left = 0
    right = len(lst) - 1
    while left < right:
        lst[left], lst[right] = lst[right], lst[left]
        left += 1
        right -= 1
    return "".join([chr(i) for i in lst])


def runTestCase(s: str) -> None:
    s1 = reverseStr(s)
    s2 = reverseList(s)
    print(f"Original: {s} Reversed BuildIn: {s1} Reversed List: {s2}")


if __name__ == "__main__":
    runTestCase('Hello, World!')
    runTestCase('Lorem Ipsum Dolor Senet')

File: interviews/lc/12-reverse-string.cpp
#include "lc-utility.h"

void reverseString(string& s) {
  if (s.length() <= 1) {
    return;
  }
  size_t left = 0, right = s.length() - 1;
  while (left < right) {
    const auto tmp = s[left];
    s[left] = s[right];
    s[right] = tmp;
    left++;
    right--;
  }
}

void runTestCase(string s) {
  cout << "Original: " << s << endl;
  reverseString(s);
  cout << "Reversed: " << s << endl;
}

int main(int argc, char* argv[]) {
  runTestCase("Hello, world!");
  runTestCase("Lorem Ipsum Dolor Senet");
  runTestCase("");
  runTestCase("a");
  return 0;
}
